src/policies.rs

```diff
@@ -375,6 +443,79 @@ impl AllocBlock {
             .correct_buffer()
     }
 
+    pub fn encrypt_buffer_ffi<'a>(w: WeakMut<'a, AllocBlock>) {
+        w.get_ref_mut()
+            .expect("encrypt_buffer_ffi")
+            .encrypt_buffer()
+    }
+
+    pub fn decrypt_buffer_ffi<'a>(w: WeakMut<'a, AllocBlock>) {
+        w.get_ref_mut()
+            .expect("decrypt_buffer_ffi")
+            .decrypt_buffer()
+    }
+
+    /// TODO: refactor encrypt and decrypt buffer
+    fn encrypt_buffer(&mut self) {
+        let mut buffer = self.buffer();
```

Comment:
Even though they are mostly equivalent, use `self.buffer_mut()`

Reply:
This doesn't work, because the borrow checker complains about all the other times it can't use self.policies because it was mutably borrowed, I'd have to change a lot of stuff.